mbf_abstract_nav/src/move_base_action.cpp

```diff
@@ -0,0 +1,590 @@
+#include "mbf_abstract_nav/move_base_action.h"
+#include <mbf_utility/navigation_utility.h>
+#include <mbf_abstract_nav/MoveBaseFlexConfig.h>
+
+namespace mbf_abstract_nav{
+
+MoveBaseAction::MoveBaseAction(const std::string &name,
+                               const RobotInformation &robot_info,
+                               const std::vector<std::string> &behaviors)
+  :  name_(name), robot_info_(robot_info), private_nh_("~"),
+     action_client_exe_path_(private_nh_, "exe_path"),
+     action_client_get_path_(private_nh_, "get_path"),
+     action_client_recovery_(private_nh_, "recovery"),
+     oscillation_timeout_(0),
+     oscillation_distance_(0),
+     recovery_enabled_(true),
+     cancel_(false),
+     exe_path_canceled_(false),
+     behaviors_(behaviors),
+     action_state_(NONE),
+     recovery_trigger_(NONE),
+     replanning_rate_(private_nh_.param<double>("replanning_frequency", 1.0))
```

Comment:
Some applications may not need the re-planning function. Whereas, ros::Rate does not allow value to be 0, or it will crash in next cycle.

Reply:
Thanks four your comments and the testing! Will check that and find a solution!